pandas/tests/resample/test_timedelta.py

```diff
@@ -105,7 +105,7 @@ def test_resample_categorical_data_with_timedeltaindex():
         index=pd.to_timedelta([0, 10], unit="s"),
     )
     expected = expected.reindex(["Group_obj", "Group"], axis=1)
-    expected["Group"] = expected["Group_obj"].astype("category")
+    expected["Group"] = expected["Group_obj"]
```

Comment:
Another test change. The expected result here is less clear I think. You could argue for either categorical or not...

Reply:
This is an API change from 0.25.x. Dunno what to do here :/

Reply:
On 0.25.x, we have the following

```python
In [42]: df = pd.DataFrame({"A": ['a', 'b']}, dtype='category', index=pd.date_range('2000', periods=2))

In [43]: df.resample("2D").agg(lambda x: 'a')
Out[43]:
   A
0  a

In [44]: df.resample("2D").agg(lambda x: 'c')
Out[44]:
     A
0  NaN
```

We're potentially changing the *values* of the result by returning a CategoricalDtype there. I don't think that behavior is correct, so I'm in favor of making this breaking change... I think...

Reply:
> We're potentially changing the values of the result by returning a CategoricalDtype there. I don't think that behavior is correct, so I'm in favor of making this breaking change... I think...

That seems exactly the bug we were fixing initially. For groupby it was a regression, so fine to do it as a fix for resample I think.

Reply:
I see now that I commented about in two different places (also in the whatsnew), with a contradicting message ...

I don't know what is best for now in this PR, but in general: if we are going to take the rule of "only trying to cast back if the scalars are of the dtype.type", then I think we should try to cast to categorical (meaning, we should see the string "a" as a valid scalar of a categorical dtype with string categories). The specific case of strings is a bit difficult of course, as also the categories' dtype is object and can hold anything.

Now, long term, I am thinking we should maybe not use this "scalar of dtype.type" rule (we can say that to explain it, but not as actual check in the code). But we could rather dictate that `_from_sequence` should be strict. And then it's `_from_sequence` that can hold the logic to know if it's a proper scalar (eg to know that pd.NA is also fine, or to know that "a" is a proper scalar for a categorical with categories that include "a", or knowing that a python int is also fine in addition to np.int64 scalar for an IntDtype, etc). 
For the specific case of categorical, it could then fail if it's getting scalars (eg "c" in the example above) which cannot be a scalar of its dtype (since the categories are part of the dtype, it can know "c" is not a valid scalar in that case). 

You will still get variable behaviour depending on what the agg function returns, eg `.agg(lambda x: 'a')` you get back categorical and `.agg(lambda x: 'c')` not, while both are simple functions that return a string. But with any rule such small differences will be unavoidable for the *default* behaviour (hence having a keyword to have full control might be interesting to investigate)

Reply:
Sorry for the long comment here :) That should probably be moved to the follow-up issue.  
About what is best for *this* PR, I am not fully sure. I suppose what you have now is fine. The `isinstance(result[0], dtype.type)` check just doesn't really work for categoricals ..

Reply:
Sorry, missed this earlier. I'm writing up the followup issue now, but will address this there.